### PR TITLE
142734455 dk create permissionsdb contract

### PR DESCRIPTION
**Pull request by DavidKnott**
Creates and test PermissionsDB contract.
Until we have Doug up and running the creator of the contract will be given admin status and then they can create other admins or basic users.
Fully tested.
@case-eee @ski-climb @tmikeschu @ethanbennett

### CODE REVIEW

File: contracts/PathsDB.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.4;
```
Comment (tmikeschu) [inline contracts/PathsDB.sol:1]: It looks like this branch was branched off of your other PR?

**Review by tmikeschu — CHANGES_REQUESTED** [review]
Looks great! I think we should either merge the other PR first, or reduce this PR to just permissions DB, since the path stuff exists in the other PR.

**Comment by DavidKnott** [comment]
@tmikeschu I closed the first PR because this one has the same stuff.  Now I think it's ready to merge.

**Comment by ethanbennett** [comment]
Looks good to me, nice work!

**Review by tmikeschu — APPROVED** [review]

File: contracts/PathsDB.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    function addCheckpoint(address checkpoint) returns (bool) {
+      if (checkpoint == 0x0) {
+        return false;
+      }
+      if (path.length == 0 || path[path.length-1] != checkpoint) {
+        path.push(checkpoint);
+        return true;
+      }
+      return false;
```
Comment (ski-climb) [inline contracts/PathsDB.sol:15]: I dig the conservative approach - return false is the general case.

File: test/TestPathsDB.sol [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  function testCheckpointCannotBeAddedIfAddressIsTheSameAsTheLastOneAndReturnsFalse(){
+    pathsDB.addCheckpoint(testAddress1);
+    Assert.equal(pathsDB.addCheckpoint(testAddress1), false, "PathsDB doesn't return false when a duplicate address is provided.");
+  }
+
+  function testCheckpointCannotBeEmptyAndReturnsFalse(){
+    Assert.equal(pathsDB.addCheckpoint(0x0), false, "PathsDB doesn't return false when an empty address is provided.");
+  }
```
Comment (ski-climb) [inline test/TestPathsDB.sol:27]: I dig the tests.  Nice job @DavidKnott

**Review by ski-climb — APPROVED** [review]

**Pull request merged**